lib/jsonapi_compliable/scope.rb

```diff
@@ -91,7 +94,25 @@ def sideload(results, includes)
           end
         else
           namespace = Util::Sideload.namespace(@namespace, sideload.name)
-          sideload.resolve(results, @query, namespace)
+          resolve_sideload = -> { sideload.resolve(results, @query, namespace) }
+          if concurrent
+            promises << Concurrent::Promise.execute(&resolve_sideload)
+          else
+            resolve_sideload.call
+          end
+        end
+      end
+
+      if concurrent
```

Comment:
Looking at the `ruby-concurrent` docs, I think you might get a more robust version of this by combining [`zip`](http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/Promise.html#zip-class_method) and [`wait!`](http://ruby-concurrency.github.io/concurrent-ruby/Concurrent/IVar.html#wait%21-instance_method):

``` ruby
all = Concurrent::Promise.zip(promises)
all.wait!(the_timeout)
```

Reply:
Nice! I'll give that a shot. I remember `.all?` wasn't working as documented :(